examples/repeat_entry.rb

```diff
@@ -11,7 +11,7 @@
 
 puts "Ok, you did it."
 
-pass = ask("<%= @key %>:  ") do |q|
+pass = ask("<%= key %>:  ") do |q|
```

Comment:
Hi @maurogeorge ,

This PR is just a "pause" to get great feedbacks as the ones you and the others are giving. But I agree, there's this "incomplete" status that makes things hard to understand.

Basically at the old code @key was resolved because almost everything was running inside the class HighLine itself. Now, I have some classes where things are fowarded/delegated to. So, now, it has to be "key", it is a method call that will be resolved to the proper receiver.

At the end, I'll update all the comments/documentation.